app/graphql/types/git/directory_entry_type.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength
+Types::Git::DirectoryEntryType = GraphQL::InterfaceType.define do
+  # rubocop:enable Metrics/BlockLength
+  name 'DirectoryEntry'
+  description 'A directory entry (directory or file) of a repository'
+
+  field :name, !types.String do
+    description 'The name of the entry'
+  end
+
+  field :path, !types.String do
+    description 'The path of the entry'
+  end
+
+  # rubocop:disable Metrics/BlockLength
+  field :log, !types[!Types::Git::CommitType] do
```

Comment:
The ordering is newest to oldest. It's true that this is the other way around compared to all other lists, but it makes sense in the context of a history. You are usually not interested in the oldest changes, but rather in the most recent changes.